### PR TITLE
Refactor set cache tests to start from IIR and add missing tests

### LINKED ISSUE #774: rework dawn unit test: dawn/Optimizer/Passes/TestPassSetCaches
starts at sir, should be finer grained base on decision taken

### PR DESCRIPTION
**Pull request by eddie-c-davis**
## Technical Description

Refactors set cache pass tests to start from IIR rather than SIR, add missing cache tests, and remove old SIR files.

### Resolves / Enhances

Put the link to the issue here
Resolves #774

### CODE REVIEW

**Comment by eddie-c-davis** [comment]
launch jenkins

File: dawn/test/unit-test/dawn/Optimizer/Passes/TestPassSetCaches.cpp [hunk 1]
```diff
@@ -17,91 +17,235 @@
...
-          ASSERT_TRUE(multiStage->getCache(accessID).getIOPolicy() == ioPolicies[j][k]);
-        }
-        j += 1;
+    ASSERT_GT(stencils.size(), 0);
+
+    auto& stencil = stencils[0];
+    auto& multiStages = stencil->getChildren();
+    ASSERT_EQ(multiStages.size(), nMultiStages);
+
+    int i = 0;
```
Comment (jdahm) [inline dawn/test/unit-test/dawn/Optimizer/Passes/TestPassSetCaches.cpp]: Could we use `enumerate` here? https://github.com/MeteoSwiss-APN/dawn/blob/master/dawn/src/dawn/Support/External/easy_iterator.hpp#L395

Comment (eddie-c-davis) [reply dawn/test/unit-test/dawn/Optimizer/Passes/TestPassSetCaches.cpp]: Done!

**Comment by eddie-c-davis** [comment]
launch jenkins

File: dawn/test/unit-test/dawn/Optimizer/Passes/TestPassSetCaches.cpp [hunk 2]
```diff
@@ -17,91 +17,230 @@
...
-          ASSERT_TRUE(multiStage->getCache(accessID).getIOPolicy() == ioPolicies[j][k]);
-        }
-        j += 1;
+    ASSERT_GT(stencils.size(), 0);
+
+    auto& stencil = stencils[0];
+    auto& multiStages = stencil->getChildren();
+    ASSERT_EQ(multiStages.size(), nMultiStages);
+
+    for(auto [i, multiStage] : enumerate(multiStages)) {
```
Comment (jdahm) [inline dawn/test/unit-test/dawn/Optimizer/Passes/TestPassSetCaches.cpp]: This looks great :)

**Pull request merged**